pinot-core/src/test/java/org/apache/pinot/queries/BaseQueriesTest.java

```diff
@@ -223,8 +223,11 @@ private BrokerResponseNative getBrokerResponse(QueryContext queryContext, PlanMa
       byte[] serializedResponse = instanceResponse.toBytes();
       dataTableMap.put(new ServerRoutingInstance("localhost", 1234, TableType.OFFLINE),
           DataTableFactory.getDataTable(serializedResponse));
-      dataTableMap.put(new ServerRoutingInstance("localhost", 1234, TableType.REALTIME),
-          DataTableFactory.getDataTable(serializedResponse));
+      // skip creating the realtime table for gapfill test case.
+      if (!queryContext.isAggregateGapfill()) {
```

Comment:
This should not be required. This test won't query the realtime table, but just mimic the behavior of hitting 2 servers

Reply:
Do you mean sqlQuery will not hit the realtime table, right? How about pqlQuery? It seems that pqlQuery will query both tables.

Reply:
Fixed.